frappe/translate.py

```diff
@@ -99,8 +99,8 @@ def get_language(lang_list: list = None) -> str:
 		if parent_language in lang_set:
 			return parent_language
 
-	# fallback to language set in User or System Settings
-	return frappe.local.lang
+	# fallback to language set in System Settings or "en"
+	return frappe.db.get_default("lang") or "en"
```

Comment:
in which scenario this fallback value is used?

Reply:
Hi Ankush, thanks for your time! The fallback is used when none of below ones are reached:
	1. Form Dict => _lang
	2. Cookie => preferred_language (Non authorized user)
	3. Request Header => Accept-Language (Non authorized user)
	4. User document => language
	5. System Settings => language
For example when you enter first time in site, no passing _lang in url, no having cookie language (because you have no select any language from language selector yet), your browser accepts only german, for example (not in enabled langs), you are not logged, so default language will be shown as first entry point :)

Reply:
Not 100% sure but browsers send accept-language headers by default pretty much always, so I think this code never gets executed 🤔 

anyway.